### PR TITLE
Support to fetch Windows service options

### PR DESCRIPTION
**Pull request by thxCode**
**Issue:**
https://github.com/rancher/rancher/issues/22470

### CODE REVIEW

File: drivers/rke/rke_driver.go [hunk 1]
```diff
@@ -180,6 +181,10 @@ func getData(s Store, k8sVersion string) map[string]interface{} {
 	if data2 != nil {
 		data["k8s-service-options"] = data2
 	}
+	data3 := s.GetWindowsServiceOptions(k8sVersion)
+	if data3 != nil {
+		data["k8s-windows-service-options"] = data3
```
Comment (kinarashah) [inline drivers/rke/rke_driver.go]: corresponding to this, RKE needs to read this from `k8s-windows-service-options`, which is done here.. https://github.com/rancher/rke/blob/master/cmd/up.go#L89 

and it gets read while building kubelet, kubeApi etc when passed in data. 
Here's an example of it reading from Kubelet: 
https://github.com/rancher/rke/blob/master//cluster/plan.go#L444

File: drivers/rke/rke_driver.go [hunk 2]
```diff
@@ -52,6 +52,7 @@ type Store interface {
 	// Add methods here to get rke driver specific data
 	GetAddonTemplates(k8sVersion string) map[string]interface{}
 	GetServiceOptions(k8sVersion string) *v3.KubernetesServicesOptions
+	GetWindowsServiceOptions(k8sVersion string) *v3.KubernetesServicesOptions
```
Comment (kinarashah) [inline drivers/rke/rke_driver.go]: I think it'd make sense to combine of them in one. The return type could be `map[string]*v3.KubernetesServicesOptions` with `"k8s-service-options"` and `"k8s-windows-service-options"` as key. 

This way, when we read it here, https://github.com/rancher/rke/blob/master//cluster/plan.go#L444, RKE changes would get easier.

Comment (thxCode) [reply drivers/rke/rke_driver.go]: Updated

File: drivers/rke/rke_driver.go (hunk 1 above)
Comment (thxCode) [reply drivers/rke/rke_driver.go]: Thanks, for now, we don't support to use `rke up` to set up a Windows cluster, it only could be created by rancher server. The purpose of placing parameters/options/images into metadata is more convenient for future unified management.

Comment (kinarashah) [reply drivers/rke/rke_driver.go]: It's not actually for `rke up`, but for rancher. When a new kubernetes version gets added with some new service options for windows in `kontainer-driver-metadata` and rancher wants to send that information to rke...we pass these options to rke. 

RKE needs to be able to read them, the line points to `ClusterUp` which is called by rancher as well as cmd. So if data has any value, we need to use that else we'll fall back to default ones.

**Comment by thxCode** [comment]
@kinarashah , got it.

**Pull request merged**